decidim-core/app/controllers/decidim/locales_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+require_dependency "decidim/application_controller"
+
+module Decidim
+  # A controller to allow users switching their locale.
+  class LocalesController < ApplicationController
+    def create
+      if current_user && params["locale"] && available_locales.include?(params["locale"])
+        current_user.update_attribute(:locale, params["locale"])
+      end
+
+      uri = URI(params["back"] || "/")
+      params = URI.decode_www_form(uri.query || "") << ["locale", current_locale]
```

Comment:
What about if we prepend it to the route, like:

`/ca`
`/ca/sign_up`

This is one of the recommended options by Google (other is top level domain names or subdomains, but those aren't options for us).